lib/evaluate.js

```diff
@@ -147,9 +149,10 @@ export const evaluate = async ({
 
   const start = Date.now()
   const buckets = createSetScoresBuckets(participants)
+  let tx
   for (const [bucketIndex, bucket] of Object.entries(buckets)) {
     try {
```

Comment:
Here is how I understand your new version at the high level:


```js
  let tx
  for (const [bucketIndex, bucket] of Object.entries(buckets)) {
    tx = await ieContractWithSigner.setScores // etc.
  }
  tx.wait()
```

This code will wait only for the last transaction created.

If that's intentional, then let's make it more obvious to the reader, please.

For example:

```js
  let lastTx
  for (const [bucketIndex, bucket] of Object.entries(buckets)) {
    const tx = await ieContractWithSigner.setScores // etc.
    lastTx = tx
  }
  // add a comment explaining why we wait for the last transaction only
  // and don't need to wait for other transactions
  lastTx.wait()
```

Reply:
This was even more broken. Each tx should be awaited and then marked as confirmed. Fixed 👍

Reply:
It may be worth creating a test to verify this works as intended. However, mocking all the dependencies to simulate two TXs running concurrently will require quite a bit of code, so it may not be worth it right now.